File: ECommerce.Domain/Entites/Product.cs
using ECommerce.Domain.Common;
using ECommerce.Domain.Enums;

namespace ECommerce.Domain.Entites
{
    public class Product : AuditableEntity<int>
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int Stock { get; set; }

        public int CategoryId { get; set; }
        public ProductStatus Status { get; set; } = ProductStatus.Available;
        public int SellerId { get; set; }
        //public SellerProfile Seller { get; set; }
        public Category Category { get; set; }
        public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
        public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
    }
}

File: ECommerce.Client/src/app/core/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  stock: number;
  categoryId: number;
  categoryName?: string;
  images?: any[];
  imageUrls?: string[];
  rating?: number;
  sellerName?: string;
  storeDes?: string;
  isDeleted:boolean;
}
